16/umbraco-cms/customizing/foundation/context-api/consume-a-context.md

```diff
@@ -1,199 +1,405 @@
 ---
 description: >-
-  Consuming a Context via the Context API is the way to start the communication
-  with the rest of the application
+  Learn how to consume contexts in Umbraco elements using one-time references
+  or subscriptions to access data and functionality through the Context API.
 ---
-
 # Consume a Context
+There are two ways to consume a context: get a __one-time reference__ to the context, or get a __subscription__ for handling context changes. The Context API is a flexible system where contexts can get disconnected or replaced. A subscription allows for the handling of these changes. However, subscriptions use more resources. They are typically consumed in the constructor, a time when the computer is already processing a lot. Which way to go depends on your use case.
 
-## Start consuming
-
-There are different ways to consume a Context API. The most straightforward implementation is done on an [Umbraco Element](../umbraco-element/) with a Context Token.
+A one-time reference approach is suitable for fire-and-forget events. The key here is that the context is not needed on initialization, but is only needed when a specific criteria is met. For instance, events that occur after user interaction or when a specific function is called. In that case, you need to get a context, do something and forget about the context after that.
 
-All Umbraco Context APIs have a Context Token which can be imported and used for consumption, for example:
-
-```typescript
-import { UMB_NOTIFICATION_CONTEXT } from '@umbraco-cms/backoffice/notification';
+If you need a context during initialization which is then set as a variable, you should always use a subscription. Otherwise you risk holding on to a context that could be disconnected or replaced without you knowing.
```

Comment:
```suggestion
If you need a context during initialization which is then set as a variable, you should always use a subscription. Otherwise, you risk holding on to a context that could be disconnected or replaced without you knowing.
```